src/components/withLocalize.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+import compose from '../libs/compose';
+import ONYXKEYS from '../ONYXKEYS';
+import {translate} from '../libs/translate';
+import DateUtils from '../libs/DateUtils';
+import {toLocalPhone, fromLocalPhone} from '../libs/LocalePhoneNumber';
+import numberFormat from '../libs/numberFormat';
+
+const withLocalizePropTypes = {
+    // Translations functions using current User's preferred locale
+    translations: PropTypes.shape({
+        translate: PropTypes.func.isRequired,
+        numberFormat: PropTypes.func.isRequired,
+        timestampToRelative: PropTypes.func.isRequired,
+        timestampToDateTime: PropTypes.func.isRequired,
+        toLocalPhone: PropTypes.func.isRequired,
+        fromLocalPhone: PropTypes.func.isRequired,
+    }),
+};
+
+function withLocalizeHOC(WrappedComponent) {
+    const withLocalize = (props) => {
+        const translations = {
```

Comment:
What is the best way here passing as an object or separately?

Reply:
What do you mean exactly? This looks good to me.

Reply:
Just asking if this looks good :sweat_smile: